core/services/ardupilot_manager/firmware/FirmwareInstall.py

```diff
@@ -0,0 +1,104 @@
+    FirmwareInstallFail,
+    InvalidFirmwareFile,
+    UndefinedPlatform,
+    UnsupportedPlatform,
+)
+from firmware.FirmwareDownload import FirmwareDownloader, FirmwareFormat, Platform
+from firmware.FirmwareUpload import FirmwareUploader
+
+
+def get_board_id(platform: Platform) -> int:
+    ardupilot_board_ids = {
+        Platform.Pixhawk1: 9,
+    }
+    return ardupilot_board_ids.get(platform, -1)
+
+
+class FirmwareInstaller:
+    """Abstracts the install procedures for different supported boards.
+
+    For proper usage one needs to set the platform before using other methods.
+
+    Args:
+        firmware_folder (pathlib.Path): Path for firmware folder.
+    """
+
+    def __init__(self, firmware_folder: pathlib.Path) -> None:
+        self._firmware_folder: pathlib.Path = firmware_folder
+
+    @staticmethod
+    def validate_firmware(firmware_path: pathlib.Path, platform: Platform) -> None:
+        """Check if given firmware is valid for given platform."""
+        if platform == Platform.Undefined:
+            raise UndefinedPlatform("Platform is undefined. Cannot validate firmware.")
+
+        if platform == Platform.Pixhawk1:
+            # For Pixhawk/Serial boards only .apj files are supported
+            try:
+                with open(firmware_path, "r") as firmware_file:
+                    firmware_data = firmware_file.read()
+                firmware_json = json.loads(firmware_data)
+                board_id = int(firmware_json.get("board_id", -1))
+                if board_id == -1:
+                    raise InvalidFirmwareFile("Could not find board_id specification in the firmware file.")
+                if board_id != get_board_id(platform):
+                    raise InvalidFirmwareFile(f"Expected board_id {get_board_id(platform)}, found {board_id}.")
+                return
+            except Exception as error:
+                raise InvalidFirmwareFile(f"Could not load firmware file for validation: {error}") from error
+
+        firmware_format = FirmwareDownloader._supported_firmware_formats[platform]
+        if firmware_format == FirmwareFormat.ELF:
+            # TODO: Include firmware validation through Ardupilot's ardupilot_fw_decoder.py
+            return
+
+        raise UnsupportedPlatform("Firmware validation is not implemented for this platform.")
+
+    @staticmethod
+    def add_run_permission(firmware_path: pathlib.Path) -> None:
+        """Add running permission for firmware file."""
+        # Make the binary executable
+        ## S_IX: Execution permission for
+        ##    OTH: Others
+        ##    USR: User
+        ##    GRP: Group
+        ## For more information: https://www.gnu.org/software/libc/manual/html_node/Permission-Bits.html
+        os.chmod(firmware_path, firmware_path.stat().st_mode | stat.S_IXOTH | stat.S_IXUSR | stat.S_IXGRP)
+
+    def install_firmware(
+        self, new_firmware_path: pathlib.Path, firmware_dest_path: pathlib.Path, platform: Platform
```

Comment:
we need better docs here. firmware_dest_path has no relevance for Pixhawk, does it? The docstring should say so. It is also not quire clear from the signature what it is for.

Reply:
Yeah...
I will make `firmware_dest_path` optional and raise in case `platform` is not Pixhawk and this argument is missing.